src/base/utils/random.cpp

```diff
@@ -29,47 +29,117 @@
 
 #include "random.h"
 
-#include <cassert>
-#include <chrono>
+#include <limits>
 #include <random>
 
 #include <QtGlobal>
 
-#ifdef Q_OS_MAC
-#include <QThreadStorage>
+#ifdef Q_OS_WIN
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+#include <Windows.h>
+#include <Ntsecapi.h>
+#else  // Q_OS_WIN
+#include <cstdio>
 #endif
 
-// on some platform `std::random_device` may generate the same number sequence
-static bool hasTrueRandomDevice{ std::random_device{}() != std::random_device{}() };
+#include "misc.h"
 
-uint32_t Utils::Random::rand(const uint32_t min, const uint32_t max)
+namespace
 {
-#ifdef Q_OS_MAC  // workaround for Apple xcode: https://stackoverflow.com/a/29929949
-    static QThreadStorage<std::mt19937> generator;
-    if (!generator.hasLocalData())
-        generator.localData().seed(
-            hasTrueRandomDevice
-            ? std::random_device{}()
-            : (std::random_device::result_type) std::chrono::system_clock::now().time_since_epoch().count()
-        );
-#else
-    static thread_local std::mt19937 generator{
-        hasTrueRandomDevice
-        ? std::random_device{}()
-        : static_cast<std::random_device::result_type>(std::chrono::system_clock::now().time_since_epoch().count())
+#ifdef Q_OS_WIN
+    class RandomLayer
+    {
+        // need to satisfy UniformRandomBitGenerator requirements
+
+        public:
+            using result_type = uint32_t;
+
+            explicit RandomLayer()
+                : m_rtlGenRandom {Utils::Misc::loadWinAPI<PRTLGENRANDOM>("Advapi32.dll", "SystemFunction036")}
```

Comment:
This seems like an internal function.
[This](https://msdn.microsoft.com/en-us/library/windows/desktop/aa387694(v=vs.85).aspx) seems to recommend the use of [CryptGenRandom](https://msdn.microsoft.com/en-us/library/windows/desktop/aa379942(v=vs.85).aspx). But it also has a deprecation notice, and I believe the new thing is [BCryptGenRandom](https://msdn.microsoft.com/en-us/library/windows/desktop/aa375458(v=vs.85).aspx). However, that is only available from Vista+.
Could I recommend using `CryptGenRandom` for the time being?

Reply:
Actually more are using `RtlGenRandom` than you thought: 
Botan use it: https://github.com/randombit/botan/pull/1494
google's boringSSL: https://boringssl.googlesource.com/boringssl/+/master/crypto/rand_extra/windows.c#44
SO has an answer: https://stackoverflow.com/q/48875929